src/Html/HtmlRuntime.fs

```diff
@@ -183,16 +191,33 @@ module HtmlRuntime =
                         if i < rows.Length && j < numberOfColumns
                         then res.[i].[j] <- data
 
+        let numberOfHeaderRows = res |> Array.countWhile (Array.forall (fun r -> r.IsHeader))
+
+        let hasRealHeaders, res = 
+            match numberOfHeaderRows with
+            | 0 -> false, res
+            | 1 -> true, res
+            | _ ->
+                for i = 1 to numberOfHeaderRows - 1 do
+                    for j = 0 to numberOfColumns - 1 do
+                        let previousCell = res.[i-1].[j]
+                        let thisCell = res.[i].[j]
+                        if previousCell.Data <> "" && thisCell.Data <> "" && thisCell.Data <> previousCell.Data then
+                            res.[i].[j] <- Cell(true, previousCell.Data + " - " + thisCell.Data)
+        
+                true, res.[numberOfHeaderRows-1..]
```

Comment:
I would prefer to leave the table structure intact and just do this at the TP level, but it was getting quite complicated, so I left it like this. @colinbull any better suggestion?